python/dgl/data/utils.py

```diff
@@ -350,3 +361,395 @@ def __len__(self):
             Number of datapoints in the subset
         """
         return len(self.indices)
+
+################### Feature Processing #######################
+
+def row_normalize(features):
+    mx = sp.csr_matrix(features, dtype=np.float32)
+
+    """Row-normalize sparse matrix"""
+    rowsum = np.array(mx.sum(1))
+    r_inv = np.power(rowsum, -1).flatten()
+    r_inv[np.isinf(r_inv)] = 0.
+    r_mat_inv = sp.diags(r_inv)
+    mx = r_mat_inv.dot(mx)
+    return np.array(mx.todense())
+
+def col_normalize(features):
+    mx = sp.csr_matrix(features, dtype=np.float32)
+
+    colsum = np.array(mx.sum(0))
+    c_inv = np.power(colsum, -1).flatten()
+    c_inv[np.isinf(c_inv)] = 0.
+    c_mat_inv = sp.diags(c_inv).transpose()
+    mx = mx.dot(c_mat_inv)
+    return np.array(mx.todense())
+
+def float_row_l1_normalize(features):
+    rowsum = np.sum(np.abs(features), axis=1)
+    r_inv = np.power(rowsum, -1).reshape(-1,1)
+    r_inv[np.isinf(r_inv)] = 0.
+    return features * r_inv
+
+def float_col_l1_normalize(features):
+    colsum = np.sum(np.abs(features), axis=0)
+    c_inv = np.power(colsum, -1)
+    c_inv[np.isinf(c_inv)] = 0.
+    return features * c_inv
+
+def embed_word2vec(str_val, nlps):
+    """ Use NLP encoder to encode the string into vector
+
+    There can be multiple NLP encoders in nlps. Each encoder
+    is invoded to generate a embedding for the input string and
+    the resulting embeddings are concatenated.
+
+    Parameters
+    ----------
+    str_val : str
+        words to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+    """
+    vector = None
+    for nlp in nlps:
+        doc = nlp(str_val)
+        if vector is None:
+            vector = doc.vector
+        else:
+            vector = np.concatenate((vector, doc.vector))
+    return vector
+
+def parse_lang_feat(str_feats, nlps, verbose=False):
+    """ Parse a list of strings using word2vec encoding using NLP encoders in nlps
+
+    Parameters
+    ----------
+    str_feats : list of str
+        list of strings to encode
+
+    nlps : list of func
```

Comment:
Why is this list of func? Maybe better to call it nlp_encoders

Reply:
Good idea